src/Grammars/Availability/AvailabilityGrammar.php

```diff
@@ -47,7 +47,29 @@ class AvailabilityGrammar extends Generator
     // coarseness is one of the above constant.
     public function generateAvailability($busyList, $ranges, $coarseness=self::BASE, $context)
     {
-        return $this->generate([ 'busyList' => $busyList, 'ranges' => $ranges, 'coarseness' => $coarseness ], $context);
+        $invalidRanges = [];
+        foreach ($busyList as $rangeIndex => $range) {
+            list($day, $startTime, $endTime) = $range;
+
+            // Convert start and end times to minutes for easier comparison
+            $startMinutes = $startTime[0] * 60 + $startTime[1];
+            $endMinutes = $endTime[0] * 60 + $endTime[1];
+
+            // Check if the start time is greater than or equal to the end time
+            if ($startMinutes >= $endMinutes) {
+                $invalidRanges[] = $rangeIndex;
+            }
+        }
+
+        if (!empty($invalidRanges)) {
+            echo "Invalid time range found in range(s): ";
+            foreach ($invalidRanges as $rangeIndex) {
+                $rangeIndex = $rangeIndex + 1;
+                echo "$rangeIndex, ";
```

Comment:
Basically change these `echo` into creating a string and using that for the `Exception`

Reply:
I have done so, thanks @DrDub